torch/distributed/fsdp/flat_param.py

```diff
@@ -2065,8 +2066,11 @@ def _uses_reduce_mixed_precision(self) -> bool:
     @property
     def _force_full_precision(self) -> bool:
         return (
-            self._training_state == HandleTrainingState.SUMMON_FULL_PARAMS
-            and self._uses_param_mixed_precision
+            (self._uses_param_mixed_precision or self._uses_reduce_mixed_precision) and (
+                self._training_state == HandleTrainingState.SUMMON_FULL_PARAMS or
+                # Also disable mixed precision in model eval mode
+                not self._fully_sharded_module.training
```

Comment:
How should we handle hybrid `eval()` and non-`eval()`? What if the parent module is in `eval()`, but the child module is not?

For a single fully sharded module per handle, taking the fully sharded module's training mode makes sense, but if we ever break that to do execution order wrapping, then what should we do then? Force full precision as long as _any_ of the parent modules is in eval mode?

Reply:
I don't see a use case for hybrid eval / non eval. Why would the user want some parts of the model to be in training and others to be eval? Unless they're doing something like disabling dropout / batchnorm tracking for only part of the model.

Reply:
If a use case for hybrid eval / non-eval comes up, we can consider additional discussions. Currently, I believe there is always one fully sharded module per handle so this check should be okay until we have execution order based wrapping?

Reply:
I agree this is not a major concern right now. I mainly was just exercising the thought :)

I think this should be fine as is.